azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java

```diff
@@ -214,6 +226,32 @@ protected String getClientId() {
         return targetAccount.login();
     }
 
+    /**
+     * see doc for: az account list-locations -o table
+     */
+    public List<Region> listRegions(String subscriptionId) {
+        return getSubscription(subscriptionId).listLocations().stream()
+                .filter(l -> l.regionType() == RegionType.PHYSICAL) // use distinct since com.azure.core.management.Region impls equals
+                .map(Location::region).distinct().map(AzureAccount::toRegion).collect(Collectors.toList());
+    }
+
+    /**
+     * see doc for: az account list-locations -o table
+     */
+    public List<Region> listRegions() {
+        return Flux.fromIterable(getSubscriptions()).parallel().map(com.microsoft.azure.toolkit.lib.common.model.Subscription::getId)
+                .map(this::listRegions)
+                .sequential().collectList()
+                .map(regionSet -> regionSet.stream()
+                .flatMap(Collection::stream)
+                .filter(Utils.distinctByKey(region -> StringUtils.lowerCase(region.getLabel()))) // cannot distinct since Region doesn't impl equals
```

Comment:
Impl equals method by label in library `Region`?

Reply:
will be moved in future PR